3rdparty/CMakeLists.txt

```
@@ -104,15 +104,16 @@ add_library(mypaint STATIC libmypaint/brushsettings-gen.h
             libmypaint/mypaint-rectangle.h libmypaint/mypaint-surface.c
             libmypaint/mypaint-surface.h libmypaint/rng-double.c
             libmypaint/rng-double.h )
-set_dp_target_properties(mypaint NO_CLANG_TIDY NO_WARNINGS)
-target_include_directories(mypaint INTERFACE
-                           "${CMAKE_CURRENT_LIST_DIR}/libmypaint")
+target_include_directories(mypaint SYSTEM PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/libmypaint>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
 
 add_library(uthash INTERFACE)
-set_dp_target_properties(uthash NO_CLANG_TIDY NO_WARNINGS)
-target_include_directories(uthash INTERFACE "${CMAKE_CURRENT_LIST_DIR}/uthash")
+target_include_directories(uthash INTERFACE "uthash")
 
-add_library(jo_gifx STATIC jo_gifx/jo_gifx.h jo_gifx/jo_gifx.c)
-set_dp_target_properties(jo_gifx NO_WARNINGS)
-target_include_directories(
-    jo_gifx INTERFACE "${CMAKE_CURRENT_LIST_DIR}/jo_gifx")
+add_library(jo_gifx OBJECT jo_gifx/jo_gifx.h jo_gifx/jo_gifx.c)
+target_include_directories(jo_gifx SYSTEM PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/jo_gifx>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
```

Currently, clang-tidy will run on all these targets, spewing lots of warnings about stuff we're not gonna fix. So I think disabling it for all targets in this directory would make sense:

```suggestion
)

get_property(third_party_targets
    DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
    PROPERTY BUILDSYSTEM_TARGETS)
set_target_properties(${third_party_targets}
    PROPERTIES C_CLANG_TIDY "" CXX_CLANG_TIDY "")
```

---

I made a slightly different change to keep all of the overrides generated in one place and applied in another one place, but it is updated now and I promise it doesn’t emit tidy warnings on third-party code now probably.